src/pages/GameDisplayPage.js

```diff
@@ -45,27 +53,99 @@ const GameDisplayPage = () => {
   //   }],
   //   isFirstMove: true,
   // });
+  let initBoard = new Array(7);
+  for (let y = 0; y < 7; y++) {
+    initBoard[y] = new Array(6).fill(null);
+  }
+
+  const [board, setBoard] = useState(initBoard);
+  const [isNextPlayerRed, setIsNextPlayerRed] = useState(false);
+  const [gameWinner, setGameWinner] = useState("");
 
-  const [board, setBoard] = useState(new Array(6).fill(Array(7).fill(null)));
+  //モーダルの開閉
+  const [open, setOpen] = useState(false);
+  // ゲームが終了した時に起動する↓
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
 
+  // ゲームの状態の初期化
   const initGame = () => {
-    let board = new Array(6).fill(Array(7).fill(null));
-    // setBoard((prevState) => board);
-    setBoard(() => board); // Yuki Ueno: エラーが出るので削除しました
+    setBoard(initBoard);
+    setGameWinner("");
+    setIsNextPlayerRed(false);
+  };
+
+  // ボードの深いコピーを作成
+  const copyBoard = (board) => {
+    let copiedBoard = [];
+    for (const array of board) {
+      copiedBoard.push([...array]);
+    }
+    return copiedBoard;
+  };
+
+  // 選択した列に石が置けるか判定
+  const canPutStone = (board, x) => {
+    if (board[x][5] != null) return false;
+    return true;
+  };
+
+  // 選んだ列の一番下の空いているy座標を返す
+  const getYIndex = (board, x) => {
+    for (let y = 0; y < 6; y++) {
+      if (board[x][y] == null) {
+        return parseInt(y);
```

Comment:
ここの`y`はfor文で0で初期化したものなので、型はかならず`number`になります！
```suggestion
        return y;
```